### DISCUSSION TITLE
some problem in websocket proxy

### DISCUSSION THREAD

**Discussion by liujun69** [apache/apisix#7559] [Q&A]
![image](https://user-images.githubusercontent.com/27419058/181448697-9c192aef-f5e3-4940-bb2a-8dac9f9d18f4.png)![image](https://user-images.githubusercontent.com/27419058/181448897-4529f3e0-5a5f-431b-ad1e-155c70538bc9.png)
The config in above. Set the cert and websocket address info in upsteam;then set the proxy path and rewrite https in route.
The  actual websocket service id wss://ip:port/ws, when visit by 'ws://127.0.0.1:9080/ws', the request reach apisix service but forward error.
2022/07/28 15:24:58 [info] 44#44: *27028 [lua] init.lua:378: http_access_phase(): matched route: {"clean_handlers":{},"createdIndex":34186,"key":"\/apisix\/routes\/418527933334815422","value":{"uri":"\/ws","priority":0,"upstream_id":"418527821128794814","name":"mqtt","enable_websocket":true,"id":"418527933334815422","update_time":1658993052,"create_time":1658991687,"status":1,"methods":["GET","POST","PUT","DELETE","PATCH","HEAD","OPTIONS","CONNECT","TRACE"]},"update_count":0,"orig_modifiedIndex":34200,"modifiedIndex":34200,"has_domain":false}, client: 172.18.0.1, server: _, request: "GET /ws HTTP/1.1", host: "172.16.15.131:9080"
2022/07/28 15:24:58 [error] 44#44: *27028 [lua] upstream.lua:555: get_by_id(): failed to find upstream by id: 418527821128794814, client: 172.18.0.1, server: _, request: "GET /ws HTTP/1.1", host: "172.16.15.131:9080"
2022/07/28 15:24:58 [info] 44#44: *27028 client 172.18.0.1 closed keepalive connection

**Comment by liujun69** [comment]
create a ws proxy is ok, but wss is error and throw failed to find upstream by id: 418527821128794814,is this means already forward but the targt service no response?

**Reply by tokers** [reply]
Could you check out the error.log more carefully? I suspect the upstream object `418527821128794814` was dropped by APISIX due to the schema error.

**Reply by liujun69** [reply]
![image](https://user-images.githubusercontent.com/27419058/181482689-724356b8-2b43-498a-90f7-2902b5552e24.png)
upstream 418527821128794814 is in the dashboard view

**Reply by tokers** [reply]
In the dashboard view doesn't represent it can be used by APISIX. Please read my last message carefully.

**Reply by tzssangglass** [reply]
check `418527821128794814` in ETCD

**Reply by liujun69** [reply]
> check `418527821128794814` in ETCD
/apisix/upstreams/
init_dir
/apisix/upstreams/418527821128794814
{"id":"418527821128794814","create_time":1658991621,"update_time":1659325364,"nodes":{"172.16.1.188:15676":1},"timeout":{"connect":30,"send":30,"read":30},"type":"roundrobin","scheme":"https","pass_host":"pass","name":"rabbit","tls":{"client_cert":"xxxx","client_key":"xxxx"},"keepalive_pool":{"idle_timeout":60,"requests":1000,"size":320}}

**Reply by liujun69** [reply]
The problem is resoved. It caused by the cert content must be started by '-----BEGIN CERTIFICATE-----' and end by '-----END CERTIFICATE-----'
